### PR TITLE
fix(test-runner): fix error when function metadata varies between tests

### PR DESCRIPTION
**Pull request by Stuk**
Fixes https://github.com/modernweb-dev/web/issues/689

See also https://github.com/istanbuljs/v8-to-istanbul/issues/121

## What I did

1. This PR updates `addingMissingCoverageItems` to also add missing functions. This can occur when a function is defined in another function, and the outer function is not called in every test run. For example methods defined on a class using arrow functions `=>` are only created when the constructor runs. Specifically, I have moved the code inside the loops out into their own functions, so they can be called with the branch coverage data and the function coverage data.

### CODE REVIEW

**Comment by Stuk** [comment]
Let me know what changeset I should add. Maybe a minor bump?

**Comment by LarsDenBakker** [comment]
@Stuk Thanks a lot! We can do a patch, you can run `yarn changeset` or add one by clicking the PR comment. I can also do it for you.

@benjamind @Westbrook  could you check if this fixes it for you as well?

**Comment by Westbrook** [comment]
Looking!

**Comment by benjamind** [comment]
Confirmed this fixes the reproducible test case found [here](https://github.com/Stuk/wtr-coverage-repro)

**Comment by Stuk** [comment]
@LarsDenBakker actually, I don't want to mess anything up, so would you ind adding the changeset?

**Comment by Westbrook** [comment]
The results certainly look nice here!

Before:
image

After:
image

File: packages/test-runner-core/src/coverage/getTestCoverage.ts [hunk 1]
```diff
@@ -26,61 +27,105 @@ export interface TestCoverage {
...
+
 /**
- * Cross references coverage mapping data, looking for missing code branches
- * and adding empty entries for them if found. This is necessary because istanbul
- * expects code branch data to be equal for all coverage entries, while v8 only
- * outputs actual covered code branches.
+ * Cross references coverage mapping data, looking for missing code branches and
+ * functions and adding empty entries for them if found. This is necessary
+ * because istanbul expects code branch and function data to be equal for all
+ * coverage entries. V8 only outputs actual covered code branchesm and functions
```
Comment (Westbrook) [inline packages/test-runner-core/src/coverage/getTestCoverage.ts:86]:
```suggestion
 * coverage entries. V8 only outputs actual covered code branches and functions
```

**Comment by Stuk** [comment]
Added changeset with some help from Westbrook. Thanks!

**Review by Westbrook — APPROVED** [review]
I think this is good to go! It's already a big win for my usage in Spectrum Web Components as outlined in the screenshots above and I know a number of other have bumped into this as well. Thanks for digging into this for us @Stuk 🎉

**Comment by LarsDenBakker** [comment]
Thanks, much appreciated!

**Pull request merged**